### PR TITLE
ENH: move file methods to utils.files

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #336 

- Moves the methods from `pysat._files` to `pysat.utils.files`.  Includes deprecation warnings for the old methods, which point to the new methods.
- Includes a flake8 sweep of `_files.py` and `test_files.py`, as well as the removal of those exceptions.

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

via pytest

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
TODO
- [ ] update documentation
- [ ] add unit tests for deprecation warnings

**Comment by jklenzing** [comment]
Docs for new api are building.  This should be good for review.

File: pysat/utils/files.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+    Note
+    ----
+        If two files have the same date and time information in the
+        filename then the file with the higher version/revision is used.
+        Series returned only has one file der datetime. Version is required
+        for this filtering, revision is optional.
+
+    """
+
+    from pysat.utils.time import create_datetime_index
```
Comment (aburrell) [inline pysat/utils/files.py:38]: All imports belong at top

File: pysat/utils/files.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    -------
+    OrderedDict
+        Information parsed from filenames
+        'year', 'month', 'day', 'hour', 'minute', 'second', 'version',
+        'revision'
+        'files' - input list of files
+
+    """
+
+    import collections
```
Comment (aburrell) [inline pysat/utils/files.py:134]: Move to top

File: pysat/utils/files.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    OrderedDict
+        Information parsed from filenames
+        'year', 'month', 'day', 'hour', 'minute', 'second', 'version',
+        'revision'
+        'files' - input list of files
+        'format_str' - formatted string from input
+
+    """
+
+    import collections
```
Comment (aburrell) [inline pysat/utils/files.py:217]: Remove this one

File: pysat/utils/files.py [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    # create storage for data to be parsed from filenames
+    stored = collections.OrderedDict()
+    stored['year'] = []
+    stored['month'] = []
+    stored['day'] = []
+    stored['hour'] = []
+    stored['minute'] = []
+    stored['second'] = []
+    stored['version'] = []
+    stored['revision'] = []
```
Comment (aburrell) [inline pysat/utils/files.py:145]:
```suggestion
    ordered_keys = ['year', 'month', 'day', 'hour', 'minute', 'second',
                    'version', 'revision']
    stored = collections.OrderedDict({kk: list() for kk in ordered_keys})
```

File: pysat/utils/files.py [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+                        else:
+                            search_str += '*'
+                        break
+            else:
+                raise ValueError("Couldn't determine formatting width")
+
+    return {'search_string': search_str,
+            'keys': keys,
+            'lengths': lengths,
+            'string_blocks': snips}
```
Comment (aburrell) [inline pysat/utils/files.py:359]:
```suggestion
    return out_dict
```

File: pysat/utils/files.py [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+            with data of length length.
+
+    Note
+    ----
+        The '?' may be used to indicate a set number of spaces for a variable
+        part of the name that need not be extracted.
+        'cnofs_cindi_ivm_500ms_{year:4d}{month:02d}{day:02d}_v??.cdf'
+
+    """
+
```
Comment (aburrell) [inline pysat/utils/files.py:316]:
```suggestion
    out_dict = {'search_string': '', 'keys': [], 'lengths': [], 'string_blocks': []}
```

File: pysat/utils/files.py [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+    if format_str is None:
+        raise ValueError("Must supply a filename template (format_str).")
+
+    # parse format string to figure out how to construct the search string
+    # to identify files in the filesystem
+    search_str = ''
```
Comment (aburrell) [inline pysat/utils/files.py:322]:
```suggestion
```

File: pysat/utils/files.py [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+    form = string.Formatter()
+    # stores the keywords extracted from format_string
+    keys = []
+    # and length of string
+    snips = []
+    lengths = []
```
Comment (aburrell) [inline pysat/utils/files.py:328]:
```suggestion
```

File: pysat/utils/files.py [hunk 9]
```diff
@@ -0,0 +1,396 @@
...
+    for snip in form.parse(format_str):
+        # collect all of the format keywords
+        # replace them in the string with the '?' wildcard
+        # numnber of ?s goes with length of data to be parsed
+        # length grabbed from format keywords so we know
+        # later on where to parse information out from
+        search_str += snip[0]
+        snips.append(snip[0])
```
Comment (aburrell) [inline pysat/utils/files.py:336]:
```suggestion
        out_dict['snips'].append(snip[0])
```

File: pysat/utils/files.py [hunk 10]
```diff
@@ -0,0 +1,396 @@
...
+    # and length of string
+    snips = []
+    lengths = []
+    for snip in form.parse(format_str):
+        # collect all of the format keywords
+        # replace them in the string with the '?' wildcard
+        # numnber of ?s goes with length of data to be parsed
+        # length grabbed from format keywords so we know
+        # later on where to parse information out from
+        search_str += snip[0]
```
Comment (aburrell) [inline pysat/utils/files.py:335]:
```suggestion
        out_dict['search_str'] += snip[0]
```

File: pysat/utils/files.py [hunk 11]
```diff
@@ -0,0 +1,396 @@
...
+        snips.append(snip[0])
+        if snip[1] is not None:
+            keys.append(snip[1])
```
Comment (aburrell) [inline pysat/utils/files.py:338]:
```suggestion
            out_dict['keys'].append(snip[1])
```

File: pysat/utils/files.py [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+            # try and determine formatting width
+            temp = re.findall(r'\d+', snip[2])
+            if temp:
+                # there are items, try and grab width
+                for i in temp:
+                    # make sure there is truly something there
+                    if i != 0:
+                        # store length and add to the search string
+                        lengths.append(int(i))
```
Comment (aburrell) [inline pysat/utils/files.py:347]:
```suggestion
                        out_dict['lengths'].append(int(i))
```

File: pysat/utils/files.py [hunk 13]
```diff
@@ -0,0 +1,396 @@
...
+                        if not wildcard:
+                            search_str += '?' * int(i)
```
Comment (aburrell) [inline pysat/utils/files.py:349]:
```suggestion
                            out_dict['search_str'] += '?' * int(i)
```

File: pysat/utils/files.py [hunk 14]
```diff
@@ -0,0 +1,396 @@
...
+                        else:
+                            search_str += '*'
```
Comment (aburrell) [inline pysat/utils/files.py:351]:
```suggestion
                            out_dict['search_str'] += '*'
```

File: pysat/utils/files.py [hunk 15]
```diff
@@ -0,0 +1,396 @@
...
+        is provided by pysat to the instrument_module.list_files
+        functions as data_path.
+    search_str : string
+        String to search local file system for
+        Ex: 'cnofs_cindi_ivm_500ms_????????_v??.cdf'
+            'cnofs_cinfi_ivm_500ms_*_v??.cdf'
+
+    Returns
+    -------
+    list
```
Comment (aburrell) [inline pysat/utils/files.py:379]:
```suggestion
    files : list
```

File: pysat/utils/files.py [hunk 16]
```diff
@@ -0,0 +1,396 @@
...
+        Supports 'year', 'month', 'day', 'hour', 'minute', 'second', 'version',
+        and 'revision'
+        Ex: 'cnofs_vefi_bfield_1sec_{year:04d}{month:02d}{day:02d}_v05.cdf'
+    wildcard : bool
+        if True, replaces the ? sequence with a * . This option may be well
+        suited when dealing with delimited filenames.
+
+    Returns
+    -------
+    dict
```
Comment (aburrell) [inline pysat/utils/files.py:298]:
```suggestion
    out_dict : dict
```

**Comment by jklenzing** [comment]
Updated docstring info for two-year break keyword (see #510)

**Pull request merged**